lib/winrm/http/transport.rb

```diff
@@ -67,6 +89,28 @@ def no_ssl_peer_verification!
         @httpcli.ssl_config.verify_mode = OpenSSL::SSL::VERIFY_NONE
       end
 
+      # SSL Peer Fingerprint Verification
+      def ssl_peer_fingerprint_verification!
+        # Not sure if this is the right place to be making
+        # a probe / tcp connection to grab the cert and compare during
+        # initialize....
+        # We could just analyize the fingerprint after first request
+        noverify_peer_context = OpenSSL::SSL::SSLContext.new
+        noverify_peer_context.verify_mode = OpenSSL::SSL::VERIFY_NONE
+        tcp_connection = TCPSocket.new(@endpoint.host, @endpoint.port)
+        shady_ssl_connection = OpenSSL::SSL::SSLSocket.new(tcp_connection,
+                                                           noverify_peer_context)
+        shady_ssl_connection.connect
```

Comment:
ssl_fingerprint_verification! will make an outbound tcp connection before sending any traffic to verify our cert fingerprint. See WinRb/WinRM#151